### ISSUE TITLE
Has anyone encountered this issue: Unable to invoke no-args constructor for class. Registering an InstanceCreator with Gson for this type may fix this problem?

### ISSUE THREAD

**Issue by beslimir** [philipplackner/CurrencyConverter#3]

**Comment by beslimir** [comment]
I tried to get the data from the (now a bit different) API call, but I got the mentioned error.
Does anyone know what's exactly the issue?
I tried this on my own way in a different app, and then just to correct Philipp's app, but both times I got the came error...

**Comment by jpunzak** [comment]
Yes, I have encountered it and cannot seem to resolve it.  I have already faced the api issue and simply converted to another exchange rate api.  Now I am facing this issue and have read a lot of resolutions but I have not been able to resolve it.  One post said remove the "suspend" when calling getRates but that did not resolve the issue.  

UPDATE:   I followed beslimir's advice and now it works!   I decreased the number of rates in the Rate class to < 65.   Of course this is just a temp fix but I'm just trying to learn so that's OK for now.  Thanks beslimir!!!

**Comment by beslimir** [comment]
Rejecting invocation, expected 65 argument registers, method signature has 66 or more...
This is the issue I got while trying to use all 160 currencies in dte data class constructor.

**In summary**: The constructor accepts 65 arguments at most, but we used here 160. I just reduced the number of currencies, but didn't research more about this issue, I don't know how to use them all.
